subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultModuleVersionIdentifier.java

```diff
@@ -21,19 +21,20 @@
 
 public class DefaultModuleVersionIdentifier implements ModuleVersionIdentifier {
 
-    private final DefaultModuleIdentifier id;
+    private final ModuleIdentifier id;
     private final String version;
 
     public DefaultModuleVersionIdentifier(String group, String name, String version) {
```

Comment:
Should we make the `DefaultModuleVersionIdentifier` private as well?

Should some of the identifiers which are created through here also be cached? (e.g. `DefaultMavenPublication.getCoordinates()`)